common/src/api/internal/shared.rs

```diff
@@ -57,6 +57,7 @@ pub struct NetworkInterface {
     pub vni: external::Vni,
     pub primary: bool,
     pub slot: u8,
+    pub transit_ips: Option<Vec<external::IpNet>>,
```

Comment:
This is `Option`al instead of just a `Vec` because this type seems to be used in stored blueprints etc., and this seemed the easiest way not to break deserialisation. This field should only be used on instance NICs.

I don't know whether we'd prefer a `#[serde(default)]` here, or something else entirely since service NICs, zone config etc. should never use this property.